src/lumigo_tracer/wrappers/http/sync_http_wrappers.py

```diff
@@ -60,17 +67,29 @@ def update_event_response(
                         the aggregated response body
     This function assumes synchronous execution - we update the last http event.
     """
-    last_event = SpansContainer.get_span().remove_last_span()
+    last_event = SpansContainer.get_span().pop_last_span()
     if last_event:
+        http_info = last_event.get("info", {}).get("httpInfo", {})
         if not host:
-            host = last_event.get("info", {}).get("httpInfo", {}).get("host", "unknown")
+            host = http_info.get("host", "unknown")
         else:
             HttpState.previous_response_body = b""
 
+        has_error = status_code >= 400
         headers = {k.lower(): v for k, v in headers.items()} if headers else {}
         parser = get_parser(host, headers)()  # type: ignore
-        if len(HttpState.previous_response_body) < Configuration.max_entry_size:
+        if len(HttpState.previous_response_body) < Configuration.get_max_entry_size(has_error):
             HttpState.previous_response_body += body
+        if has_error and HttpState.previous_request and http_info.get("request"):
+            max_size = Configuration.get_max_entry_size(True)
+            http_info["request"].update(
+                {
+                    "body": lumigo_dumps(HttpState.previous_request.body, max_size)
```

Comment:
what are you trying to do?
You're using the requested body in the response?
Why did you added this logic?